src/parser/Literal.cpp

```diff
@@ -149,4 +149,11 @@ void Literal::setSubstr(std::size_t start, std::size_t length) {
 // __________________________________________
 void Literal::removeDatatypeOrLanguageTag() { content_.erase(beginOfSuffix_); }
 
+// __________________________________________
+void Literal::replaceContentWithSameLength(const std::string& newContent) {
+  AD_CORRECTNESS_CHECK(newContent.size() == beginOfSuffix_ - 2);
+  for (std::size_t i = 1; i < beginOfSuffix_ - 1; ++i) {
+    content_[i] = newContent[i - 1];
+  }
+}
```

Comment:
As stated above, you need an arbitrary `replaceContent` function, 
that automatically inserts or deletes characters if the string is longer or shorter.